### PR TITLE
Return 404 in shareinfo.php when outgoing s2s disabled

### LINKED ISSUE #13578: Unknown error when access a remote share after remote sharing has been disabled
### Steps to reproduce

1. Install owncloud on two systems
2. on system one, remote share a file with a user on system two
3. on system two, accept share and validate access
4. on system one, as the admin user uncheck the "Allow users on this server to send shares to other servers" checkbox
5. on system two, access the share

### Expected behaviour

Unknown:  Should the remote user still have access to the share?

### Actual behaviour

An "unknown error" is thrown.

p253

### Server log file

```

{"reqId":"a0d1a685efd816381e0d1a1f61a4bfee","remoteAddr":"192.168.1.218","app":"files","message":"Exception: {\"Message\":\"The passed data is not valid XML\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Client.php(181): Sabre\\\\DAV\\\\Client->parseMultiStatus(false)\\n#1 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/storage\\\/dav.php(513): Sabre\\\\DAV\\\\Client->propFind('', Array)\\n#2 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files_sharing\\\/lib\\\/external\\\/storage.php(144): OC\\\\Files\\\\Storage\\\\DAV->hasUpdated('', 1421875827)\\n#3 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/wrapper.php(357): OCA\\\\Files_Sharing\\\\External\\\\Storage->hasUpdated('', 1421875827)\\n#4 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/wrapper.php(357): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Wrapper->hasUpdated('', 1421875827)\\n#5 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/cache\\\/watcher.php(67): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Wrapper->hasUpdated('', 1421875827)\\n#6 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/view.php(932): OC\\\\Files\\\\Cache\\\\Watcher->checkUpdate('', Array)\\n#7 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/filesystem.php(782): OC\\\\Files\\\\View->getFileInfo('\\\/picdir', true)\\n#8 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files\\\/ajax\\\/list.php(12): OC\\\\Files\\\\Filesystem::getFileInfo('\\\/picdir')\\n#9 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/route\\\/route.php(135) : runtime-created function(1): require_once('\\\/var\\\/www\\\/html\\\/o...')\\n#10 [internal function]: __lambda_func(Array)\\n#11 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/route\\\/router.php(250): call_user_func('?lambda_1697', Array)\\n#12 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/base.php(763): OC\\\\Route\\\\Router->match('\\\/apps\\\/files\\\/aja...')\\n#13 \\\/var\\\/www\\\/html\\\/owncloud\\\/index.php(28): OC::handleRequest()\\n#14 {main}\",\"File\":\"\\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Client.php\",\"Line\":540}","level":4,"time":"2015-01-21T21:37:10+00:00"}


```

### Server Host info:

Ubuntu 14.04
PHP 5.5.9

### Server configuration

ownCloud Enterprise Edition 8.0 beta 1 (daily) Build:2015-01-21T04:13:37+00:00

### PR DESCRIPTION
**Pull request by PVince81**
Fixes https://github.com/owncloud/core/issues/13578

Please review @schiesbn @nickvergessen @icewind1991

### CODE REVIEW

**Comment by karlitschek** [comment]
makes sense :+1:

File: apps/files_sharing/ajax/shareinfo.php [hunk 1]
```diff
@@ -28,6 +28,11 @@
 	exit;
 }
 
+if (OCA\Files_Sharing\Helper::isOutgoingServer2serverShareEnabled() === false) {
+	\OC_Response::setStatus(404); //400 Bad Request
```
Comment (schiesbn) [inline apps/files_sharing/ajax/shareinfo.php]: comment doesn't match the actual value :wink: 

But makes sense... :+1:

**Comment by PVince81** [comment]
Let me fix the comment, then I'll merge

**Comment by PVince81** [comment]
Let me force push, just to break Jenkins. You know, for fun.

**Comment by PVince81** [comment]
The mighty Jenkins PR: https://github.com/owncloud/core/pull/16612

**Comment by PVince81** [comment]
@schiesbn's thumbs up is hidden in the old commit, where I forgot to change the commit.
Merging hence.

**Pull request merged**